jetty-ee10/jetty-ee10-servlet/src/test/java/org/eclipse/jetty/ee10/servlet/DefaultServletCombinationsTest.java

```diff
@@ -0,0 +1,191 @@
+import jakarta.servlet.http.HttpServletResponse;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.jetty.http.HttpTester;
+import org.eclipse.jetty.server.HttpConfiguration;
+import org.eclipse.jetty.server.LocalConnector;
+import org.eclipse.jetty.server.ResourceService;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.toolchain.test.FS;
+import org.eclipse.jetty.toolchain.test.jupiter.WorkDir;
+import org.eclipse.jetty.toolchain.test.jupiter.WorkDirExtension;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.eclipse.jetty.server.ResourceService.WelcomeMode.REDIRECT;
+import static org.eclipse.jetty.server.ResourceService.WelcomeMode.REHANDLE;
+import static org.eclipse.jetty.server.ResourceService.WelcomeMode.SERVE;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+@ExtendWith(WorkDirExtension.class)
+public class DefaultServletCombinationsTest
+{
+    public WorkDir workDir;
+
+    public Path docRoot;
+    private Server server;
+    private LocalConnector connector;
+    private ServletContextHandler context;
+
+    @BeforeEach
+    public void init() throws Exception
+    {
+        docRoot = workDir.getEmptyPathDir().resolve("docroot");
+        FS.ensureDirExists(docRoot);
+        Path subdirPath = docRoot.resolve("subdir");
+        FS.ensureDirExists(subdirPath);
+        Files.writeString(docRoot.resolve("index.html"), "Static index.html at root", UTF_8);
+        Files.writeString(docRoot.resolve("foo.welcome"), "Static foo.welcome at root", UTF_8);
+        Files.writeString(subdirPath.resolve("index.html"), "Static index.html at subdir", UTF_8);
+        Files.writeString(subdirPath.resolve("foo.welcome"), "Static foo.welcome at subdir", UTF_8);
+
+        server = new Server();
+
+        connector = new LocalConnector(server);
+        connector.getConnectionFactory(HttpConfiguration.ConnectionFactory.class).getHttpConfiguration().setSendServerVersion(false);
+
+        context = new ServletContextHandler();
+        context.setBaseResourceAsPath(docRoot);
+        context.setContextPath("/");
+        context.setWelcomeFiles(new String[]{"index.html", "index.welcome"});
+
+        ServletHolder welcomeExtHolder = context.addServlet(WelcomeServlet.class, "*.welcome");
+        welcomeExtHolder.setInitParameter("mapping", "welcome extension");
+    }
+
+    private void startServer(boolean pathInfoOnly, ResourceService.WelcomeMode welcomeMode) throws Exception
+    {
+        ServletHolder defaultHolder = context.addServlet(DefaultServlet.class, "/");
+        defaultHolder.setInitParameter("pathInfoOnly", String.valueOf(pathInfoOnly));
```

Comment:
This is not really testing the pathInfoOnly, because there is no servletPath.   
If pathInfoOnly is true, then the DefaultServlet needs to be registered at something like "/static/*" and all the URIs from the test need to include that prefix.